### DISCUSSION TITLE
WaveSpeed_c3 is not a number or infinity

### DISCUSSION THREAD

**Discussion by EmreDumlu-UM** [Anura3D/Anura3D_OpenSource#108] [General]
Dear All,

I am working on a 2D MPM-MPI double-point formulation. I intended to define two different soil types in the domain. The first one is the dam while the second one is the base (foundation). However, when the water touches the dam particles, Anura-3D gives me the error. Is there anyone who has an idea about the error?

Materials:
![material](https://github.com/Anura3D/Anura3D_OpenSource/assets/125078884/24de7e5c-6270-4a17-ab75-9a10ada2e293)

Error:
![error_cmd](https://github.com/Anura3D/Anura3D_OpenSource/assets/125078884/b2c1c2f1-cecb-40f8-a0c0-6822f7ff085b)

Element 1050:
![element 1050](https://github.com/Anura3D/Anura3D_OpenSource/assets/125078884/080e1704-0a84-4327-997b-ddfda1165714)

Zoomed Element 1050:
![element 1050_zoom](https://github.com/Anura3D/Anura3D_OpenSource/assets/125078884/1e132407-5d2a-4335-b4a9-a75e5c821a72)

Best regards,
Emre

**Comment by Shachen75** [comment]
I have encountered the same error message for Column Collapse problem using 2 Material Point formulation.

**Reply by ayerrocolom** [reply]
This problem occurs because the code cannot get a wave speed number required to assign the minimum time step of the explicit calculation. This is usually because some material parameters (e.g., density, porosities, hydraulic conductivity) are not well defined in the GOM file.

Best,
Alba

**Reply by EmreDumlu-UM** [reply]
Dear Dr. Alba,

After trying several combinations of base materials, I then specify the body material to the base. However, ANURA3D still gives the same error. The body material properties were taken by Liang et al. (2022) (https://doi.org/10.1016/j.sandf.2020.06.004). They performed an overtopping simulation by using ANURA3D. In the tutorial or literature, I could not run into a simulation that used two kinds of soil materials in double-point.

So far, I have initiated the simulation by applying the K0 procedure, but now I started the simulation by applying gravity loading. Unfortunately, there is no difference between et al.

I appreciate your additional feedback. Thanks in advance.

Best,
Emre

**Reply by yang-geo22** [reply]
Hello, would you like to ask if you solved this problem? I'm having the same issue

**Reply by murphy-john** [reply]
Have you tried using the MPM-MP integration instead of the MPM-MIXED integration?
